packages/fuel-streams/src/streams/stream.ts

```diff
@@ -0,0 +1,163 @@
+import {
+  AckPolicy,
+  type Consumer,
+  type ConsumerConfig,
+  type ConsumerInfo,
+  DeliverPolicy,
+  StorageType,
+  type StreamState,
+} from '@nats-io/jetstream';
+import type { KV, KvEntry, KvWatchOptions } from '@nats-io/kv';
+import type { QueuedIterator } from '@nats-io/nats-core';
+import type { NatsClient } from '../client/natsClient';
+import type { StreamData } from './streamData';
+
+export interface SubscribeConsumerConfig {
+  filterSubjects: Array<string>;
+  deliverPolicy: DeliverPolicy;
+}
+
+export abstract class Streameable<T> {
+  NAME: string;
+  WILDCARD_LIST: string[];
+  abstract encode(): Promise<Uint8Array>;
+  abstract decode(encoded: Uint8Array): Promise<StreamData<T>>;
+}
+
+export class Stream<T> {
+  private store: KV;
+  private client: NatsClient;
+  private static instance: Stream<undefined>;
+
+  private constructor(store: KV, client: NatsClient) {
+    this.store = store;
+    this.client = client;
+  }
+
+  public static async getOrInit<T>(
+    client: NatsClient,
+    streamable: Streameable<T>,
+  ): Promise<Stream<T>> {
+    if (!Stream.instance) {
+      Stream.instance = await Stream.new<unknown>(client, streamable);
+    }
+    return Stream.instance as Stream<T>;
+  }
+
+  private static async new<T>(
+    client: NatsClient,
+    streamable: Streameable<T>,
+  ): Promise<Stream<T>> {
+    const namespace = client.getClientOpts().getNamespace();
+    const bucketName = namespace.streamName(streamable.NAME);
+    const store = await client.getOrCreateKvStore(bucketName, {
+      storage: StorageType.File,
+      history: 1,
+      compression: true,
+    });
+    return new Stream<T>(store, client);
+  }
+
+  public getStore(): KV {
+    return this.store;
+  }
+
+  private prefixFilterSubject(subject: string): string {
+    return `$KV.*.${subject}`;
```

Comment:
This implementation is an hack based on the rust `asyn_nats`'s key-value store implementation. I imagine this being very peculiar to that library. Is it by any chance the same behavior with `@nats-io`?